generated_windows.go

```diff
@@ -106,6 +106,14 @@ type (
 		// Since: generic-worker 5.3.0
 		ChainOfTrust bool `json:"chainOfTrust,omitempty"`
 
+		// Runs commands as Administrator. Property `task.payload.osGroups`
+		// must include `"Administrators"`. Requires scope
+		// `generic-worker:run-as-administrator:<provisionerId>/<workerType>`.
```

Comment:
I'm curious why there's a separate scope here.  Is it to be extra certain that the command truly does want to run as administrator?  Would we ever include Administrators in the osGroups but not let them run as administrator?  It seems to be that osGroups is already scope-protected.

Reply:
It is a great question! It turns out the `Administrators` group can be renamed, so there is no guarantee that if a group called `Administrators` exists, that it has administrative privileges. Funnily enough [Microsoft recommends renaming the Administrators group](https://technet.microsoft.com/en-us/library/cc700835.aspx?f=255&MSPPError=-2147217396#XSLTsection124121120120) (gotta love indirection, right?!).

Another subtle distinction is that new groups can be created with a subset of administrative privileges, so `runAsAdministrator` would in that case enable the administrative privileges available in the groups of the user, rather than all administrative privileges of the `Administrators` group, if it isn't a member of the Administrators group. In other words, as far as I understand, a user can still have Administrative privileges even without being a member of the `Administrators` group, or whatever it is renamed to.

We are planning to extend the UAC integration to allow a `sudo`-like feature where there is a file (similar to `/etc/sudoers` on linux/unix) that allows a whitelist of protected executables to be executed with administrative privileges. The `runAsAdministrator` feature is intended to be shortlived until the new `sudo`-like feature is available.

Reply:
wow!  yah, I think separate scopes is good in that case.